pallets/settlement/src/benchmarking.rs

```diff
@@ -360,25 +360,26 @@ pub fn setup_conditions<T: Trait>(
     dids: Vec<IdentityId>,
     ticker: Ticker,
 ) -> Vec<Condition> {
-    let mut conditions = Vec::with_capacity(count as usize);
-    for i in 0..count {
-        let scope = Scope::Custom((ticker.encode(), vec![1; i.try_into().unwrap()]).encode());
-        let claim = Claim::Jurisdiction(CountryCode::AF, scope);
-        for did in &dids {
-            identity::Module::<T>::base_add_claim(
-                did.clone(),
-                claim.clone(),
-                trusted_issuer.issuer,
-                None,
-            );
-        }
-        let condition = Condition::new(
-            ConditionType::IsPresent(claim),
-            vec![trusted_issuer.clone()],
-        );
-        conditions.push(condition);
-    }
-    conditions
+    let encoded_ticker = ticker.encode();
+    (0..count)
+        .map(|i| {
+            let scope =
+                Scope::Custom((encoded_ticker.clone(), vec![1; i.try_into().unwrap()]).encode());
+            let claim = Claim::Jurisdiction(CountryCode::AF, scope);
+            for did in &dids {
+                identity::Module::<T>::base_add_claim(
+                    did.clone(),
+                    claim.clone(),
+                    trusted_issuer.issuer,
+                    None,
+                );
+            }
+            Condition::new(
+                ConditionType::IsPresent(claim),
+                vec![trusted_issuer.clone()],
+            )
+        })
+        .collect::<Vec<_>>()
```

Comment:
Not necessary to specify the type here, though I guess you did that out of habit :)
(No need to change anything here, just noting)